src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java

```diff
@@ -0,0 +1,78 @@
+package com.internet.shop.web.filters;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthorizationFilter implements Filter {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
+    private Map<String, Set<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/orders/admin", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/admin/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/add", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/admin", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/admin/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/users/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/cart/products", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/cart/products/add", Set.of(Role.RoleName.USER));
```

Comment:
Interesting is it possible to reach cart and specifically - button "add" (or "delete"product from cart) if admin has no access to cart?

Reply:
So far as I understood from the home task text, ADMINs do not have shopping carts, only USERs do. In my program, ADMIN can add & delete products in the list of all products, not in the shopping cart)